src/toolkits/object_detection/object_detector.hpp

```diff
@@ -180,14 +185,16 @@ class EXPORT object_detector: public ml_model_base {
       const std::vector<std::pair<float, float>>& anchor_boxes,
       float min_confidence);
 
-  virtual variant_map_type perform_evaluation(gl_sframe data,
-                                              std::string metric);
+  virtual variant_type perform_evaluation(gl_sframe data, std::string metric,
+                                          std::string output_type,
+                                          float confidence_threshold,
+                                          float iou_threshold);
 
   void perform_predict(
       gl_sframe data,
       std::function<void(const std::vector<neural_net::image_annotation>&,
                          const std::vector<neural_net::image_annotation>&)>
-          consumer);
+          consumer, float confidence_threshold = 0.001, float iou_threshold = 0.45);
```

Comment:
Let's not use defaults here. Calls to `perform_predict` can be explicit. This way these magic numbers appear in fewer redundant places